File: back/Context/Models/GenericModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace SGE.Context.Models
{
    public class GenericModel
    {
        [Key]
        public long Id { get; set; }

        public bool Ativo { get; set; }

        public DateTimeOffset DataCriacao { get; set; }

        public DateTimeOffset? DataModificacao { get; set; }

    }
}

File: front/src/app/models/genericModel.ts
export class GenericModel {
    id: any;
    ativo: boolean;
    dataCriacao: Date;
    dataModificacao: Date;

    constructor(item?) {
        if (item) {
            this.id = item.id;
            this.ativo = item.ativo;
            this.dataCriacao = item.dataCriacao;
            this.dataModificacao = item.dataModificacao;
        }
    }
}
